src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -273,18 +335,917 @@ private static IEnumerable<SyntaxNode> GetOwningNodes(SyntaxNode root, int posit
+            SyntaxNode selectedNode,
+            SyntaxNode newNode,
+            ImmutableArray<SyntaxNode> nodesToInsert,
+            CancellationToken cancellationToken)
+        {
+            var rootEditor = new SyntaxEditor(root, document.Project.Solution.Workspace);
+
+            rootEditor.InsertAfter(selectedNode, nodesToInsert);
+            rootEditor.ReplaceNode(selectedNode, newNode.WithAdditionalAnnotations(s_replacementNodeAnnotation));
+            var newRoot = rootEditor.GetChangedRoot();
+
+            var newNodeAfterInsertion = newRoot.GetAnnotatedNodes(s_replacementNodeAnnotation).Single();
+            var formattingNode = newNodeAfterInsertion.FirstAncestorOrSelf<MemberDeclarationSyntax>() ?? newNodeAfterInsertion;
+            var formattedNewRoot = Formatter.Format(
+                newRoot,
+                formattingNode.Span,
+                document.Project.Solution.Workspace,
+                cancellationToken: cancellationToken);
+            var nextCaretPosition = formattedNewRoot.GetAnnotatedTokens(s_openBracePositionAnnotation).Single().Span.End;
+            return (formattedNewRoot, nextCaretPosition);
+        }
+
+        private static SyntaxNode ReplaceNodeAndFormat(
+            Document document,
+            SyntaxNode root,
+            SyntaxNode oldNode,
+            SyntaxNode newNode,
+            CancellationToken cancellationToken)
+        {
+            var annotatedNewNode = newNode.WithAdditionalAnnotations(s_replacementNodeAnnotation);
+            var newRoot = root.ReplaceNode(
+                oldNode,
+                annotatedNewNode);
+            var newNodeAfterInsertion = newRoot.GetAnnotatedNodes(s_replacementNodeAnnotation).Single();
+
+            var formattingNode = newNodeAfterInsertion.FirstAncestorOrSelf<MemberDeclarationSyntax>() ?? newNodeAfterInsertion;
+            var options = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var formattedNewRoot = Formatter.Format(
+                newRoot,
+                formattingNode.Span,
+                document.Project.Solution.Workspace,
+                options,
+                cancellationToken: cancellationToken);
+            return formattedNewRoot;
+        }
+
+        private static ObjectCreationExpressionSyntax WithArgumentListIfNeeded(ObjectCreationExpressionSyntax objectCreationExpressionNode)
+        {
+            var argumentList = objectCreationExpressionNode.ArgumentList;
+            var hasArgumentList = argumentList != null && !argumentList.IsMissing;
+            if (!hasArgumentList)
+            {
+                // Make sure the trailing trivia is passed to the argument list
+                // like var l = new List\r\n =>
+                // var l = new List()\r\r
+                var typeNode = objectCreationExpressionNode.Type;
+                var newArgumentList = SyntaxFactory.ArgumentList().WithTriviaFrom(typeNode);
```

Comment:
this is weird.  it will grab the leading trivia of the type node.  it should only grab the trailing trivia right?

Reply:
Changed it to trailing triva.